spec/lib/court_data_adaptor/resource/base_spec.rb

```diff
@@ -76,6 +76,21 @@ class CourtDataAdaptor::Resource::MockResource < described_class
             raise_error CourtDataAdaptor::Errors::BadRequest, 'Bad request'
         end
       end
+
+      context 'when unprocessable entity response' do
+        before do
+          stub_request(:get, mock_resource_endpoint)
+            .to_return(
+              status: 422,
+              body: { field: %w[error1 error2] }.to_json
+            )
+        end
+
+        it 'applies custom handler' do
+          expect { described_class.all }.to \
+            raise_error CourtDataAdaptor::Errors::BadRequest, 'Bad request'
```

Comment:
how about `Unprocessable entity` as an error message instead of `Bad request` ?

Reply:
The class name `BadRequest` might also be misleading

Reply:
@laurentqro I have created error class UnprocessableEntity. I have also changed name of the class BadRequestHandler to ApiRequestHandler, as this class will now be responsible for handling both BadRequest (400) responses and UnprocessableEntitity (422) responses. As suggested, an UnprocessableEntity response will now have a message associated 'Unprocessable entity' (although currently we are not displaying this). What do you think?